core/amber/src/main/scala/edu/uci/ics/texera/web/resource/WorkflowResultService.scala

```diff
@@ -0,0 +1,217 @@
+package edu.uci.ics.texera.web.resource
+
+import com.fasterxml.jackson.annotation.{JsonTypeInfo, JsonTypeName}
+import com.fasterxml.jackson.databind.node.ObjectNode
+import edu.uci.ics.amber.engine.architecture.controller.ControllerEvent.WorkflowResultUpdate
+import edu.uci.ics.amber.engine.architecture.principal.OperatorResult
+import edu.uci.ics.amber.engine.common.tuple.ITuple
+import edu.uci.ics.texera.web.model.event.TexeraWebSocketEvent
+import edu.uci.ics.texera.web.resource.WorkflowResultService.{
+  PaginationMode,
+  SetDeltaMode,
+  SetSnapshotMode,
+  WebOutputMode,
+  WebPaginationUpdate,
+  WebResultUpdate,
+  defaultPageSize,
+  webDataFromTuple
+}
+import edu.uci.ics.texera.web.resource.WorkflowResultService.calculateDirtyPageIndices
+import edu.uci.ics.texera.web.resource.WorkflowWebsocketResource.send
+import edu.uci.ics.texera.workflow.common.workflow.WorkflowCompiler
+import edu.uci.ics.texera.workflow.operators.sink.SimpleSinkOpDesc
+import edu.uci.ics.texera.workflow.common.IncrementalOutputMode.{SET_DELTA, SET_SNAPSHOT}
+import edu.uci.ics.texera.workflow.common.tuple.Tuple
+
+import javax.websocket.Session
+import scala.collection.mutable
+
+object WorkflowResultService {
+
+  val defaultPageSize: Int = 10
```

Comment:
I wonder if this page size can be tuned in anyway from the front end? otherwise, this is more like a fixed page size (rather than a default). 

IMO 10 is quite small as well, increasing the size might be more helpful to make the result panel more useful.

Reply:
tuning page size requires more changes, don't want to make this very complicated.